sisl/geom/surfaces.py

```diff
@@ -0,0 +1,587 @@
+            vacuum = SuperCell([0, 0, vacuums[iv] - dx])
+            iv += 1
+            out = out.add(vacuum)
+        else:
+            out = out.append(geom, 2)
+
+    if geoms_none[-1] is None:
+        # ensure that nsc is 1 if there has been inserted vacuum
+        out.set_nsc(c=1)
+
+    return out
+
+
+@set_module("sisl.geom")
+def fcc_slab(alat, atoms, miller, layers=None, vacuum=20., *, orthogonal=False, start=None, end=None):
+    r""" Construction of a surface slab from a face-centered cubic (FCC) crystal
+
+    The slab layers are stacked along the :math:`z`-axis. The default stacking is the first
+    layer as an A-layer, defined as the plane containing an atom at :math:`(x,y)=(0,0)`.
+
+    Parameters
+    ----------
+    alat : float
+        lattice constant of the fcc crystal
+    atoms : Atom
+        the atom that the crystal consists of
+    miller : int or str or (3,)
+        Miller indices of the surface facet
+    layers : int or str, optional
+        Number of layers in the slab or explicit layer specification.
+        An empty character `' '` will be denoted as a vacuum slot, see examples.
+        Currently the layers cannot have stacking faults.
+    vacuum : float or array_like, optional
+        distance added to the third lattice vector to separate
+        the slab from its periodic images. If this is None, the slab will be a fully
+        periodic geometry but with the slab layers. Useful for appending geometries together.
+        If an array layers should be a str, it should be no longer than the number of spaces
+        in `layers`. If shorter the last item will be repeated (like `zip_longest`).
+    orthogonal : bool, optional
+        if True returns an orthogonal lattice
+    start : int or string, optional
+        sets the first layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+    end : int or string, optional
+        sets the last layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+
+    Examples
+    --------
+    111 surface, starting with the A layer
+    >>> fcc_slab(alat, atoms, "111", start=0)
+
+    111 surface, starting with the B layer
+    >>> fcc_slab(alat, atoms, "111", start=1)
+
+    111 surface, ending with the B layer
+    >>> fcc_slab(alat, atoms, "111", end='B')
+
+    111 surface, with explicit layers in a given order
+    >>> fcc_slab(alat, atoms, "111", layers='BCABCA')
+
+    111 surface, with (1 Ang vacuum)BCA(2 Ang vacuum)ABC(3 Ang vacuum)
+    >>> fcc_slab(alat, atoms, "111", layers=' BCA ABC ', vacuum=(1, 2, 3))
+
+    111 surface, with (20 Ang vacuum)BCA
+    >>> fcc_slab(alat, atoms, "111", layers=' BCA', vacuum=20)
+
+    111 surface, with (2 Ang vacuum)BCA(1 Ang vacuum)ABC(1 Ang vacuum)
+    >>> fcc_slab(alat, atoms, "111", layers=' BCA ABC ', vacuum=(2, 1))
```

Comment:
Why not `vacuum=(2, 1, 1)`? Can you at least add an explanation how the third vacuum distance is determined? Otherwise this example is a bit confusing, I think.

Actually, wouldn't it not be more precise to circle through the vacuum list, say `layers='A B C A B C ' , vacuum=(1, 2)` would give alternating vacuum distances of 1 and 2 Ang?

Reply:
I have described this in the documentation, but yeah, should be emphasized in this example.
>  If shorter the last item will be repeated (like `zip_longest`).